### PR TITLE
Svelte: implement preview panel extension for match iteration

### PR DESCRIPTION
**Pull request by camdencheek**
This implements the static highlights panel extension for Codemirror in Svelte, allowing a user to click through the matched ranges in the preview panel.

## Test plan

Added a playwright test for selection navigation buttons. 

Manual testing, short video demonstration below:

https://github.com/sourcegraph/sourcegraph/assets/12631702/2ba18811-0451-41e3-8470-2f6e7ed50f66

### CODE REVIEW

File: client/web-sveltekit/src/lib/CodeMirrorBlob.svelte [hunk 1]
```diff
@@ -32,19 +32,29 @@
 
     const defaultTheme = EditorView.theme({
```
[the remark this answers is not in the document]
Comment (fkling) [reply client/web-sveltekit/src/lib/CodeMirrorBlob.svelte:33]: I assume the answer is yes: Does the file page still fully scroll with these changes?

File: client/web-sveltekit/src/lib/codemirror/static-highlights.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                ranges: this.ranges,
+                handlePrevious: this.navigatePrevious.bind(this),
+                handleNext: this.navigateNext.bind(this),
+            })
+        }
+    }
+
+    public destroy(): void {
+        this.root.$destroy()
+    }
```
[the remark this answers is not in the document]
Comment (fkling) [reply client/web-sveltekit/src/lib/codemirror/static-highlights.ts:164]: > but the component lifetime stuff got quite complex and I think this ended up quite clean.

You could have separate Svelte and React implementations of the panel, and share everything. The panel could be a configuration option on `staticHighlights`.

But not a big deal either way.

**Review by fkling — APPROVED** [review]

File: client/web-sveltekit/src/lib/codemirror/static-highlights.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (camdencheek) [reply client/web-sveltekit/src/lib/codemirror/static-highlights.ts:164]: > The panel could be a configuration option on staticHighlights.

Yeah, that's what I attempted. I couldn't figure out a good interface that could be implemented by both a react component and a svelte component.

File: client/web-sveltekit/src/lib/CodeMirrorBlob.svelte (hunk 1 above)
[the remark this answers is not in the document]
Comment (camdencheek) [reply client/web-sveltekit/src/lib/CodeMirrorBlob.svelte:33]: Yep! Tested that specifically.

**Pull request merged**